test/pkg_autocorrects_test.cc

```diff
@@ -188,17 +209,63 @@ TEST_CASE("Add test import to package with neither imports nor exports") {
                         "end\n";
 
     string expected = "class Opus::MyPackage < PackageSpec\n"
-                      "  test_import Opus::MyPackage\n"
+                      "  test_import Opus::ExamplePackage\n"
                       "end\n";
 
     auto test = TestPackageFile::create(gs, "my_package/__package.rb", pkg_source);
 
-    auto &package = gs.packageDB().getPackageForFile(gs, test.fileRef);
+    auto &package = test.targetPackage(gs);
     ENFORCE(package.exists());
-    auto addImport = package.addImport(gs, package, true);
+    auto addImport = package.addImport(gs, test.newPackage(gs), true);
     ENFORCE(addImport, "Expected to get an autocorrect from `addImport`");
     auto replaced = addImport->applySingleEditForTesting(pkg_source);
     CHECK_EQ(expected, replaced);
 }
 
+TEST_CASE("Simple add export") {
+    core::GlobalState gs(errorQueue);
+    gs.initEmpty();
+
+    string pkg_source = "class Opus::MyPackage < PackageSpec\n"
+                        "  export Opus::MyPackage::This\n"
+                        "end\n";
+
+    string expected = "class Opus::MyPackage < PackageSpec\n"
+                      "  export Opus::MyPackage::This\n"
+                      "  export Opus::MyPackage::NewExport\n"
+                      "end\n";
+
+    auto test = TestPackageFile::create(gs, "my_package/__package.rb", pkg_source);
+
+    auto &package = test.targetPackage(gs);
+    ENFORCE(package.exists());
+    auto addExport = package.addExport(gs, test.getName(gs, {"Opus", "MyPackage", "NewExport"}), false);
+    ENFORCE(addExport, "Expected to get an autocorrect from `addImport`");
+    auto replaced = addExport->applySingleEditForTesting(pkg_source);
+    CHECK_EQ(expected, replaced);
+}
+
+TEST_CASE("Simple add export_for_test") {
+    core::GlobalState gs(errorQueue);
+    gs.initEmpty();
+
+    string pkg_source = "class Opus::MyPackage < PackageSpec\n"
+                        "  export Opus::MyPackage::This\n"
+                        "end\n";
+
+    string expected = "class Opus::MyPackage < PackageSpec\n"
+                      "  export Opus::MyPackage::This\n"
+                      "  export_for_test Opus::MyPackage::NewExport\n"
+                      "end\n";
+
+    auto test = TestPackageFile::create(gs, "my_package/__package.rb", pkg_source);
+
+    auto &package = test.targetPackage(gs);
+    ENFORCE(package.exists());
+    auto addExport = package.addExport(gs, test.getName(gs, {"Opus", "MyPackage", "NewExport"}), true);
+    ENFORCE(addExport, "Expected to get an autocorrect from `addImport`");
+    auto replaced = addExport->applySingleEditForTesting(pkg_source);
+    CHECK_EQ(expected, replaced);
+}
+
```

Comment:
Why are these tests in C++ instead of a CLI test?

Reply:
I wrote these without actually wiring up the autocorrects to the code that will eventually call them, and I also wanted tests that exercised only the autocorrects without also relying on constant resolution to find the right import. A CLI test would have required the whole end-to-end be present, and also would rely on a larger number of moving parts.